xarray/tests/test_backends.py

```diff
@@ -4668,3 +4668,24 @@ def test_extract_zarr_variable_encoding():
         actual = backends.zarr.extract_zarr_variable_encoding(
             var, raise_on_invalid=True
         )
+
+
+@requires_h5netcdf
+def test_load_single_value_h5netcdf(tmp_path):
+    """Test that numeric single-element vector attributes are handled fine.
+
+    At present (h5netcdf v0.8.1), the h5netcdf exposes single-valued numeric variable
+    attributes as arrays of length 1, as oppesed to scalars for the NetCDF4
+    backend.  This was leading to a ValueError upon loading a single value from
+    a file, see #4471.  Test that loading causes no failure.
+    """
+    ds = xr.Dataset(
+        {
+            "test": xr.DataArray(
+                np.array([0]), dims=("x",), attrs={"scale_factor": 1, "add_offset": 0}
```

Comment:
How about 

```suggestion
                np.array([0]), dims=("x",), attrs={"scale_factor": [1], "add_offset": [0]}
```

and then test the output of `xr.decode_cf`?

Reply:
I like @dcherian's suggestion for a unit-test, but this test could still make sense as an integration test.

Reply:
`xr.decode_cf(ds)` doesn't fail when applied directly to an in-memory dataset.  As far as I've been able to reproduce, the problem only occurs when data are loaded lazily.  I checked the value for `data` as returned by `unpack_for_decoding(variable)` and used in `CFScaleOffsetCoder`:

- With the test I added, the value is `LazilyOuterIndexedArray(array=<xarray.backends.h5netcdf_.H5NetCDFArrayWrapper object at 0x7f0dfcc24d60>, key=BasicIndexer((slice(None, None, None),)))`, and the problem occurs.
- With the suggestion by @dcherian, the value of `data` is `array([0])`, the problem does not occur.
- I tried to replace `np.array([0])` by `indexing.LazilyOuterIndexedArray(np.array([0]))`, then the value of `data` is `LazilyOuterIndexedArray(array=NumpyIndexingAdapter(array=array([0])), key=BasicIndexer((slice(None, None, None),)))`, but the problem does not occur.

As far as I can tell, the problem only occurs with a `LazilyOuterIndexedArray` wrapping a `H5NetCDFArrayWrapper` object.  I don't know how to create or mock such an object without writing a NetCDF file and then reading it with the `h5netcdf` engine.

Reply:
Interesting. Lets just stick with your current version then. Thanks for checking.